### ISSUE TITLE
questions regarding the project

### ISSUE THREAD

**Issue by tiagom62** [thinkyhead/MarlinConfigurator#1]
What need to be done to move this project along?

**Comment by thinkyhead** [comment]
@tiagom62 With a $500 donation I could afford to put in 20-30 hours and make it really great (kidding, mostly). 😄 

Because I know exactly what I want to do, but it needs a lot of time, love, and attention.

**Comment by thinkyhead** [comment]
@tiagom62 With a $500 donation I could afford to put in 20-30 hours and make it really great (kidding… mostly). 😄 

Because I know exactly what I want to do, but it needs a lot of time, love, and attention – all in short supply right now. Most of the hooks are in place for reading and parsing the configs, generating downloads, etc., but I need to do a few more routines to make it more intelligent and bring the existing hooks up to date with the newest config formatting (particularly Doxygen-style comment blocks).

**Comment by tiagom62** [comment]
I ask because i would like to volunteer some of my time to the project. I couldn't find any documents regarding what functionality or bug fixes need implementing. I believe it would be beneficial if we could add some basic documentation of what needs to be done so that other could potentially help with the project.

I will take a look at updating the existing hooks to the new config formatting. 

I'd like to confirm that this is the offical repo that we should submit PR against.

**Comment by thinkyhead** [comment]
> I'd like to confirm that this is the offical repo that we should submit PR against.

Yes. This is the official repo for the configurator. I don't have any documentation, per se, but I did write a "part 1" article on the subject last year: http://www.thinkyhead.com/blogs/thinkyhead/2015-03-06/marlin-configurator-part-1 which gives a basic overview.

Thank you for offering your help!

The main application code is in the [`src/js/configurator.js`](https://github.com/thinkyhead/MarlinConfigurator/blob/master/src/js/configurator.js) file, so I recommend getting a good understanding of what it's doing before jumping in.

Please feel free to take the next steps, making it work with the most recent config files. The tool already handles basic dependencies, where options depend on something else being enabled. But it does not yet handle the case where an option is specified more than once (where such cases still exist – see `ALLEN_KEY_PROBE` options as an example).

The utility depends on some annotations in the config files, so feel free to propose other kinds of annotations to the configs if they seem necessary to make sense of things.

In terms of priority for Marlin, this tool might not be the most vital thing to work on, because it is attempting to be an "all in one" and deal with all possible future changes to configs. It might make more sense to imitate one of the other Marlin configuration tools out there, where the forms are more basic, the advanced options are hidden, and the final output could be configs for different versions of Marlin.
